src/cairoUtilFuncGen/storage/mappingIndexAccess.ts

```diff
@@ -218,38 +213,25 @@ export class MappingIndexAccessGen extends CairoUtilFuncGenBase {
     return {
       name: funcName,
       code: [
-        `func ${helperFuncName}{pedersen_ptr : HashBuiltin*, range_check_ptr, syscall_ptr : felt*}(`,
-        `    name : felt, ptr : felt*, len : felt`,
-        `){`,
-        `    alloc_locals;`,
-        `    if (len == 0){`,
-        `        return ();`,
+        `fn ${helperFuncName}(name: felt, ref ptr: Array::<felt>, index: u256, len: u256){`,
+        `    if index == len{`,
+        `        return;`,
         `    }`,
-        `    let index = len - 1;`,
-        `    let (index256) = felt_to_uint256(index);`,
-        `    let (loc) = ${arrayName}.read(name, index256);`,
-        `    let (value) = WARP_STORAGE.read(loc);`,
-        `    assert ptr[index] = value;`,
-        `    ${helperFuncName}(name, ptr, index);`,
-        `    return ();`,
+        `    let loc = ${arrayName}::read(name, index);`,
+        `    let value = WARP_STORAGE::read(loc);`,
+        `    ptr.append(value);`,
+        `    ${helperFuncName}(name, ref ptr, index + 1, len)`,
         `}`,
-        `func ${funcName}{pedersen_ptr : HashBuiltin*, range_check_ptr, syscall_ptr : felt*}(`,
-        `    name : felt`,
-        `) -> (hashedValue : felt){`,
-        `    alloc_locals;`,
-        `    let (len256) = ${lenName}.read(name);`,
-        `    let (len) = narrow_safe(len256);`,
-        `    let (ptr) = alloc();`,
-        `    ${helperFuncName}(name, ptr, len);`,
-        `    let (hashValue) = string_hash(len, ptr);`,
-        `    return (hashValue,);`,
+        `fn ${funcName}(name: felt) -> felt {`,
+        `    let len: u256 = ${lenName}::read(name);`,
+        `    let mut ptr = ArrayTrait::new();`,
+        `    ${helperFuncName}(name, ref ptr, 0, len);`,
+        `    string_hash(len, ptr)`,
         `}`,
       ].join('\n'),
       functionsCalled: [
-        this.requireImport('warplib.maths.utils', 'narrow_safe'),
-        this.requireImport('warplib.maths.utils', 'felt_to_uint256'),
-        this.requireImport('starkware.cairo.common.alloc', 'alloc'),
-        this.requireImport('warplib.string_hash', 'string_hash'),
+        // this.requireImport(['array'], 'ArrayTrait'),
```

Comment:
Why is it commented? I think it requires both `Array` and `ArrayTrait` to work

Reply:
It is commented because that syntax is not supported yet, it would throw a compilation error